interfaces/builtin/networkmanager.go

```diff
@@ -0,0 +1,409 @@
+				send_interface="org.freedesktop.NetworkManager.WiMax.Nsp"/>
+		<allow send_destination="org.freedesktop.NetworkManager"
+				send_interface="org.freedesktop.NetworkManager.AccessPoint"/>
+
+		<!-- Devices (read-only, no security required) -->
+		<allow send_destination="org.freedesktop.NetworkManager"
+				send_interface="org.freedesktop.NetworkManager.Device.WiMax"/>
+
+		<!-- Devices (read/write, secured with PolicyKit) -->
+		<allow send_destination="org.freedesktop.NetworkManager"
+				send_interface="org.freedesktop.NetworkManager.Device.Wireless"/>
+		<allow send_destination="org.freedesktop.NetworkManager"
+				send_interface="org.freedesktop.NetworkManager.Device"/>
+
+		<!-- Core stuff (read-only properties, no methods) -->
+		<allow send_destination="org.freedesktop.NetworkManager"
+				send_interface="org.freedesktop.NetworkManager.Connection.Active"/>
+		<allow send_destination="org.freedesktop.NetworkManager"
+				send_interface="org.freedesktop.NetworkManager.DHCP4Config"/>
+		<allow send_destination="org.freedesktop.NetworkManager"
+				send_interface="org.freedesktop.NetworkManager.DHCP6Config"/>
+		<allow send_destination="org.freedesktop.NetworkManager"
+				send_interface="org.freedesktop.NetworkManager.IP4Config"/>
+		<allow send_destination="org.freedesktop.NetworkManager"
+				send_interface="org.freedesktop.NetworkManager.IP6Config"/>
+		<allow send_destination="org.freedesktop.NetworkManager"
+				send_interface="org.freedesktop.NetworkManager.VPN.Connection"/>
+
+		<!-- Core stuff (read/write, secured with PolicyKit) -->
+		<allow send_destination="org.freedesktop.NetworkManager"
+				send_interface="org.freedesktop.NetworkManager"/>
+		<allow send_destination="org.freedesktop.NetworkManager"
+				send_interface="org.freedesktop.NetworkManager.Settings"/>
+		<allow send_destination="org.freedesktop.NetworkManager"
+				send_interface="org.freedesktop.NetworkManager.Settings.Connection"/>
+
+		<!-- Agents; secured with PolicyKit.  Any process can talk to
+				the AgentManager API, but only NetworkManager can talk
+				to the agents themselves. -->
+		<allow send_destination="org.freedesktop.NetworkManager"
+				send_interface="org.freedesktop.NetworkManager.AgentManager"/>
+		<deny send_interface="org.freedesktop.NetworkManager.SecretAgent"/>
+
+		<!-- Root-only functions -->
+		<deny send_interface="org.freedesktop.NetworkManager" send_member="SetLogging"/>
+		<deny send_interface="org.freedesktop.NetworkManager" send_member="Sleep"/>
+		<deny send_interface="org.freedesktop.NetworkManager.Settings" send_member="LoadConnections"/>
+		<deny send_interface="org.freedesktop.NetworkManager.Settings" send_member="ReloadConnections"/>
+		<deny send_interface="org.freedesktop.NetworkManager.VPN.Plugin"/>
+		<deny send_interface="org.freedesktop.NetworkManager.PPP"/>
+
+		<deny own="org.freedesktop.NetworkManager.dnsmasq"/>
+		<deny send_destination="org.freedesktop.NetworkManager.dnsmasq"/>
```

Comment:
In general I think using the upstream's bus policy is fine here. However, there are a couple of places where the policy references PolicyKit, but PolicyKit is not on Ubuntu Core (it is on sdoc though, but the interaction with a network-manager snap on an sdoc system is currently undefined (ie, two network managers would attempt to run on the system bus which is certainly not what we want)). What will network-manager do if policykit is simply not available? Furthermore we don't have a strong concept of users on Ubuntu Core so the 'default' section is overly complicated. My preference would be to take the 'root' section from upstream and cull the allow rules from the 'default' section (leaving only deny rules), unless someone can come up with a use case for the non-root user to access network manager without policy kit.

@zyga and @niemeyer , this is another point we need to consider-- some snaps will simply not work or cause problems on an sdoc system.